include/SDL3/SDL_surface.h

```diff
@@ -104,6 +115,8 @@ typedef struct SDL_Surface
     /** clipping information */
     SDL_Rect clip_rect;         /**< Read-only */
 
+    SDL_ScaleMode scaleMode;    /**< The scale mode */
```

Comment:
This might cause problems with sdl2-compat. @sezero?

Reply:
Yes, SDL3_Surface, cannot be casted as SDL2_Surface. so that would crash.
Ynless, this is put at the end of the structure.

But that would fail if we create a SDL_Surface with SDL2_image, and feeds it to sdl2-compat.

should we use the new properties for that !?
ortherwise sdl2-compat would need to convert all SDL2_surface to SDL3_surface

Reply:
> This might cause problems with sdl2-compat. @sezero?

Yes it will

>  Ynless, this is put at the end of the structure.
> 
> But that would fail if we create a SDL_Surface with SDL2_image, and feeds it to sdl2-compat.

And yes, too

Reply:
> > Ynless, this is put at the end of the structure.
> > But that would fail if we create a SDL_Surface with SDL2_image, and feeds it to sdl2-compat.
> 
> And yes, too

maybe not always, in fact. only if SDL2_image were statically linked with SDL2.

BTW. in SDL3, SDL_surface->user_data, was replaced SDL_surface->props, this may also be an issue is the application is using "user_data".